index-scheduler/src/lib.rs

```diff
@@ -407,6 +430,60 @@ impl IndexScheduler {
         Ok(this)
     }
 
+    fn index_budget(
+        tasks_path: &Path,
+        base_map_size: usize,
+        mut task_db_size: usize,
+        max_index_count: usize,
+    ) -> IndexBudget {
+        let budget = utils::dichotomic_search(base_map_size, |map_size| {
+            Self::is_good_heed(tasks_path, map_size)
```

Comment:
You should probably return an error here if it is not due to an index memory map size issue.

Reply:
See above